docs/permissions.md

```diff
@@ -0,0 +1,419 @@
+
+### `selector`
+
+By default, the `values` are checked with the `id`. But it's possible to use a
+`selector` to filter on another `field`. In particular, it can be used for
+sharing. A user may share a calendar and all the events inside it. It will be
+done with two permissions. The first one is for the calendar:
+
+```json
+{
+  "type": "io.cozy.calendars",
+  "verbs": "GET",
+  "values": ["1355812c-d41e-11e6-8467-53be4648e3ad"]
+}
+```
+
+And the other is for the events inside the calendar:
+
+```json
+{
+  "type": "io.cozy.events",
+  "verbs": "GET",
+  "selector": "calendar-id",
+  "values": ["1355812c-d41e-11e6-8467-53be4648e3ad"]
+}
+```
+
+
+## What format for a permission?
+
+### JSON
+
+The prefered format for permissions is JSON. Each permission is a map with the
+`type`, `verbs`, `values` and `selector` see above, plus a `description` that
+can be used to give more informations to the user. Only the `type` field is
+mandatory.
+
+In the manifest, the permissions are regrouped in a map. The key is not very
+relevant, it's just here for localization. The same key is used in the
+`locales` field to identify the permission.
+
+Example:
+
+```json
+{
+  "permissions": {
+    "contacts": {
+      "description": "Required for autocompletion on @name",
+      "type": "io.cozy.contacts",
+      "verbs": "GET"
+    },
+    "images": {
+      "description": "Required for the background",
+      "type": "io.cozy.files",
+      "access": "GET",
+      "values": ["io.cozy.files.music-dir"]
+    },
+    "mail": {
+      "description": "Required to send a congratulations email to your friends",
+      "type": "io.cozy.jobs",
+      "selector": "worker",
+      "values": ["sendmail"]
+    }
+  }
+}
+```
+
+### Inline
+
+OAuth2 as a `scope` parameter for defining the permissions given to the
+application. But it's only a string, not a JSON. In that case, we use a space
+delimited list of permissions, each permission is written compactly with `:`
+between the components.
+
+Example:
+
+```
+io.cozy.contacts io.cozy.files:GET:io.cozy.files.music-dir io.cozy.jobs:POST:sendmail:worker
+```
+
+**Note**: the `verbs` component can't be omitted when the `values` and
+`selector` are used.
+
+### Inspiration
+
+- [Access control on other similar platforms](https://news.ycombinator.com/item?id=12784999)
+
+
+## Routes
+
+### GET /permissions/self
+
+List the permissions for a given token
+
+#### Request
+
+```http
+GET /permissions/self HTTP/1.1
+Host: cozy.example.net
+Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ
+Accept: application/vnd.api+json
+```
+
+#### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": {
+    "type": "io.cozy.permissions",
+    "id": "5a9c1844-d427-11e6-ab36-2b684d437b0d",
+    "attributes": {
+      "application-id": "4cfbd8be-8968-11e6-9708-ef55b7c20863",
```

Comment:
Yes, I think it is safer. The slug can be reused by another application. You are OK with that?